airflow/models/dag.py

```diff
@@ -2509,6 +2524,67 @@ def bulk_write_to_db(cls, dags: Collection["DAG"], session=NEW_SESSION):
 
         DagCode.bulk_sync_to_db(filelocs, session=session)
 
+        from airflow.models.dataset import Dataset
+        from airflow.models.dataset_dag_ref import DatasetDagRef
+        from airflow.models.dataset_task_ref import DatasetTaskRef
```

Comment:
If we define the relationships right in SQLA I feel that we shouldn't need to use these classes directly, and instead should be able do do something like `dag_model.outlet_datasets:

```python
dag_model.outlet_datasets.append(DatasetModel(...))
```

I can't find the right page in the SQLA docs that talks specifically about this but https://docs.sqlalchemy.org/en/14/orm/basic_relationships.html#association-object does show it.

Reply:
Roger.  I reckon let's leave that as an optimization and not let it block this first pass?

Reply:
This one isn't  an optimisation in my head, but (I think) being able to not add 100-200 lines of code!

Reply:
I see so you are thinking that DatasetTaskRef could be like the Association object in the example, and Dataset would be like Child.  It seems like it would work with DatasetDagRef.  But with DatasetTaskRef, since we don't have a Task table, we might not be able to use this approach.... But also, a lot of those lines are for extracting the dataset references from tasks in the dag.  Anyway, may be something we can simplify somehow or another.